### DISCUSSION TITLE
running worker on windows and uploading to S3

### DISCUSSION THREAD

**Discussion by 5hai** [inlife/nexrender#674] [Q&A]
Hello

Is the "S3 provider" is already bundled within the worker  binary release version for windows  ? 

I am running a job, in the "postrender" action, i am trying to upload to S3 and this error :
`@nexrender/action-upload: Error: Couldn't find module s3, Unknown provider given.`

Job details:
`{
    "template": {
        "src": "file:///C:/Users/me/Project1.aep",
        "composition": "main_comp"
    },
    "actions": {
        "postrender": [
            {
                "module": "@nexrender/action-encode",
                "preset": "mp4",
                "output": "processing.mp4"
            },
            {
                "module": "@nexrender/action-upload",
                "input": "processing.mp4",
                "provider": "s3",
                "params": {
                    "region": "us-east-1",
                    "bucket": "testbucket",
                    "key": "folder/output.mp4",
                    "acl": "public-read",
                    "contentType": "video/mp4",
                    "credentials": {
                        "profile": "MY_PROFILE_NAME"
                    }
                }
            }
        ]
    }
}`

**Comment by Pilskalns** [comment]
Hi,

You have to install the following dependency: `@nexrender/provider-s3`

**Reply by 5hai** [reply]
I ended up running the worker via node and not using the .EXE worker 
Installed the the required dependencies.
